### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by rpire**
## Changes made:
- Create `app.rb` to handle the app's operations.
- Refactor `main.rb` to handle user's input.
- Refactor `create_person` to avoid repetition of code, following the DRY principle.
- Refactor `create_rental` to use the previously made methods of `display_books` and `display_people`, and avoid repetition of code.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    print 'Name --> '
+    name = gets.chomp
+    case decision
+    when 1
+      create_student(age, name)
+    else
+      create_teacher(age, name)
+    end
+    puts
+  end
```
Comment (o-rashad) [inline app.rb]: From the method name `create_person`, i should predict its responsible for : 
 - creating a person (student / teacher) instance.

But actually what its doing are : :thinking: 
 - line 28 to 37 : validate (and getting) user input.
 - lines 39, 41 : asking for user inputs

So what we need to do here (as requested in the project [requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md#:~:text=pay%20special%20attention%20to%20the%20%22single-responsiblity%20principle%22.%20)) is to make sure our method more simple :1234: 
-  follow the [single-responsiblity principle](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)
- all requests for input and input validation should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input.

Suggestion on how :question: 
- Extract those code snippets from your `app` module methods and put it in separate methods at `main` file.
- Then pass the inputs returned from these new methods to your `app` module methods

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    puts 'Please, enter book information below:'
+    print 'Title --> '
+    title = gets.chomp
+    print 'Author --> '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts
+    puts 'Book created successfully'
+    puts
+  end
```
Comment (o-rashad) [inline app.rb]: Same as previous comment. :arrow_up:
`create_book` method should only contain line 59

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    display_people
+    print INPT_MSG
+    person_index = gets.chomp.to_i
+    puts
+    print 'Enter date of retrieval --> '
+    date = gets.chomp
+    Rental.new(date, @books[book_index], @people[person_index])
+    puts 'Rental created successfully'
+    puts
+  end
```
Comment (o-rashad) [inline app.rb]: Same as previous comment. :arrow_up:

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    print 'Enter the person\'s ID --> '
+    display_people
+    person_id = gets.chomp.to_i
+    renter = @people.select { |person| person.id == person_id }
+    puts
+    puts "Rentals of #{renter.first.name}:"
+    puts 'EMPTY' if renter.first.rentals.empty?
+    puts(renter.first.rentals.map { |rental| "Book: #{rental.book.title}, Rented on: #{rental.date}" })
+    enter_msg
+  end
```
Comment (o-rashad) [inline app.rb]: Same as previous comment. :arrow_up:

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  private
+
+  def create_student(age, name)
+    print 'Has parent permission? [Y/N] --> '
+    permission = gets.chomp.upcase
+    permission = permission != 'N'
+    @people << Student.new(age, name, permission)
+    puts 'Student created successfully'
+    puts
+  end
```
Comment (o-rashad) [inline app.rb]: Same as previous comment. :arrow_up:

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+  def create_teacher(age, name)
+    print 'Specialty --> '
+    specialty = gets.chomp
+    @people << Teacher.new(specialty, age, name)
+    puts
+    puts 'Teacher created successfully'
+    puts
+  end
```
Comment (o-rashad) [inline app.rb]: Same as previous comment. :arrow_up:

**Review by o-rashad — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Asslam Alikom @rpire , @ReshyResh  👋😄,
I'm Omar and I will be your reviewer for this project 😊

Good job so far!👌😃

There are some issues 🐞 that you still need to work on to go to the next project but you are almost there! 🏁🏁🏁

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @rpire 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rpire** [comment]
Hello, @helman101 👋

Thank you for reviewing our work. We will proceed to merge these changes into the `dev` branch. 🚀

Happy coding to you too! 🤓

**Pull request merged**